misaligned.js

```diff
@@ -1,16 +1,51 @@
-const {expect} = require('chai')
+const { expect } = require('chai');
 
-function print_color_map() {
-    const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
-    const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
-    for (let i = 0; i < majorColors.length; i++) {
-        for (let j = 0; j < minorColors.length; j++) {
-            console.log(`${i * 5 + j} | ${majorColors[i]} | ${minorColors[j]}`);
-        }
-    }
-    return majorColors.length * minorColors.length;
+const majorColors = ['White', 'Red', 'Black', 'Yellow', 'Violet'];
+const minorColors = ['Blue', 'Orange', 'Green', 'Brown', 'Slate'];
+
+function printColorMap(
+	primaryColor,
+	secondaryColor,
+	printColors,
+	getColorsLength
+) {
+	printColors(primaryColor, secondaryColor);
+	let colorCount = getColorsLength(primaryColor, secondaryColor);
+	return colorCount;
+}
+
+function printColor(primaryColor, secondaryColor) {
+	for (let i = 0; i < primaryColor.length; i++) {
+		for (let j = 1; j < secondaryColor.length; j++) {
+			let colorIndex = getColorIndex(primaryColor.length, i, j);
+			getFormattedColorString(colorIndex, primaryColor[i], secondaryColor[j]);
+		}
+	}
+}
+
+function getColorsCount(primaryColor, secondaryColor) {
+	return primaryColor.length * secondaryColor.length;
+}
+
+function getColorIndex(
+	primaryColorLength,
+	primaryColorIndex,
+	secondaryColorIndex
+) {
+	return primaryColorIndex * primaryColorLength + secondaryColorIndex + 1;

```

Comment:
secondaryColorLength should have been used instead of primaryColorLength? please check.